cadc-util/src/main/java/ca/nrc/cadc/db/mappers/JdbcMapUtil.java

```diff
@@ -0,0 +1,339 @@
+        }
+        throw new UnsupportedOperationException("converting " + o.getClass().getName() + " " + o + " to Long");
+    }
+
+    public static UUID getUUID(ResultSet rs, int col)
+            throws SQLException {
+        Object o = rs.getObject(col);
+        if (o == null) {
+            return null;
+        }
+        if (o instanceof UUID) {
+            return (UUID) o;
+        }
+        if (o instanceof Long) {
+            return new UUID(0L, (Long) o);
+        }
+        if (o instanceof Number) {
+            // LSB
+            return new UUID(0L, ((Number) o).longValue());
+        }
+        if (o instanceof byte[]) {
+            byte[] b = (byte[]) o;
+            if (b.length < 16) {
+                // sybase truncates trailing 0s
+                byte[] bb = new byte[16];
+                System.arraycopy(b, 0, bb, 0, b.length);
+                b = bb;
+            }
+            long msb = HexUtil.toLong(b, 0);
+            long lsb = HexUtil.toLong(b, 8);
+
+            return new UUID(msb, lsb);
+        }
+        throw new UnsupportedOperationException("converting " + o.getClass().getName() + " " + o + " to UUID");
+    }
+
+    public static Float getFloat(ResultSet rs, int col)
+            throws SQLException {
+        Object o = rs.getObject(col);
+        if (o == null) {
+            return null;
+        }
+        if (o instanceof Float) {
+            return (Float) o;
+        }
+        if (o instanceof Number) {
+            return ((Number) o).floatValue();
+        }
+        throw new UnsupportedOperationException("converting " + o.getClass().getName() + " " + o + " to Float");
+    }
+
+    public static Double getDouble(ResultSet rs, int col)
+            throws SQLException {
+        Object o = rs.getObject(col);
+        return objectToDouble(o);
+    }
+
+    private static Double objectToDouble(Object o) {
+        if (o == null) {
+            return null;
+        }
+        if (o instanceof Double) {
+            return (Double) o;
+        }
+        if (o instanceof Number) {
+            return ((Number) o).doubleValue();
+        }
+        throw new UnsupportedOperationException("converting " + o.getClass().getName() + " " + o + " to Double");
+    }
+    
+    private static Long objectToLong(Object o) {
+        if (o == null) {
+            return null;
+        }
+        if (o instanceof Long) {
+            return (Long) o;
+        }
+        if (o instanceof Number) {
+            return ((Number) o).longValue();
+        }
+        throw new UnsupportedOperationException("converting " + o.getClass().getName() + " " + o + " to Long");
+    }
+
+    public static String[] getTextArray(ResultSet rs, int col)
+            throws SQLException {
+        Object o = rs.getObject(col);
+        if (o == null) {
+            return null;
+        }
+        if (o instanceof Array) {
+            Array a = (Array) o;
+            String[] ao = (String[]) a.getArray();
```

Comment:
This cast always succeeds?

Reply:
text arrays are only viable/supported in postgresql and it always works  if this method is applied to the right column type... but I think I will improve the error handling because misusing this method will likely just return a null rather than fail and that's hard to figure out.